package/src/frontend/VideoPlayer/prebuffering.js

```diff
@@ -26,30 +25,30 @@ export const prebuffering = function(player) {
   player.prebuffer = function(options) {
     options = options || {};
 
-    var delta = options.secondsToBuffer || 10;
-    var secondsToWait = options.secondsToWait || 3;
-    var interval = 200;
-    var maxCount = secondsToWait * 1000 / interval;
-    var count = 0;
-    var deferred = $.Deferred();
-    var timeout;
+    const delta = options.secondsToBuffer || 10;
+    const secondsToWait = options.secondsToWait || 3;
+    const interval = 200;
+    const maxCount = secondsToWait * 1000 / interval;
+    const promise = new Promise(() => {});
+    let count = 0;
+    let timeout;
 
     if (browser.has('prebuffering support')) {
-      if (!player.isBufferedAhead(delta) && !player.prebufferDeferred) {
+      if (!player.isBufferedAhead(delta) && !player.prebufferPromise) {
         log('prebuffering video ' + player.src());
 
         timeout = function() {
           setTimeout(function() {
-            if (!player.prebufferDeferred) {
+            if (!player.prebufferPromise) {
               return;
             }
 
             count++;
 
             if (player.isBufferedAhead(delta) || count > maxCount) {
               log('finished prebuffering video ' + player.src());
-              deferred.resolve();
-              player.prebufferDeferred = null;
+              promise.resolve();
```

Comment:
We should make sure that everything we rewrite at least has a very basic test. This for example does not exist.